app/models/bookmark.rb

```diff
@@ -10,6 +10,41 @@ class Bookmark < ActiveRecord::Base
     if: -> { reminder_type.present? && reminder_type != Bookmark.reminder_types[:at_desktop] }
   }
 
+  validate :unique_per_post_for_user
+  validate :ensure_sane_reminder_at_time
+
+  def unique_per_post_for_user
+    existing_bookmark = Bookmark.find_by(user_id: user_id, post_id: post_id)
+    return if existing_bookmark.blank? || existing_bookmark.id == id
+    self.errors.add(:base, I18n.t("bookmarks.errors.already_bookmarked_post"))
+  end
+
+  def ensure_sane_reminder_at_time
+    return if reminder_at.blank?
+    if reminder_at < Time.now.utc
```

Comment:
```suggestion
    if reminder_at < Time.zone.now
```
should be in UTC and we are moving to `Time.zone.now` in our app everywhere.